includes/data-events/listeners.php

```diff
@@ -100,3 +100,104 @@ function( $provider, $email, $lists_added, $lists_removed, $result ) {
 		];
 	}
 );
+
+/**
+ * For when there's a new donation through WooCommerce.
+ */
+Data_Events::register_listener(
+	'newspack_donation_order_processed',
+	'donation_new',
+	function( $order_id, $product_id ) {
+		$order = \wc_get_order( $order_id );
+		if ( ! $order ) {
+			return;
+		}
+		return [
+			'user_id'       => $order->get_customer_id(),
+			'email'         => $order->get_billing_email(),
+			'amount'        => (float) $order->get_total(),
+			'currency'      => $order->get_currency(),
+			'recurrence'    => get_post_meta( $product_id, '_subscription_period', true ),
+			'platform'      => 'newspack',
+			'platform_data' => [
+				'order_id'   => $order_id,
+				'product_id' => $product_id,
+			],
+		];
+	}
+);
+
+/**
+ * For when there's a new donation through the Stripe platform.
+ */
+Data_Events::register_listener(
+	'newspack_new_donation_woocommerce',
+	'donation_new',
+	function( $order, $client_id ) {
+		$order_id = $order->get_id();
+		return [
+			'user_id'       => $order->get_customer_id(),
+			'email'         => $order->get_billing_email(),
+			'amount'        => (float) $order->get_total(),
+			'currency'      => $order->get_currency(),
+			'recurrence'    => get_post_meta( $product_id, '_subscription_period', true ),
+			'platform'      => 'stripe',
+			'platform_data' => [
+				'order_id'   => $order_id,
+				'product_id' => \Newspack\Donations::get_order_donation_product_id( $order_id ),
+				'client_id'  => $client_id,
+			],
+		];
+	}
+);
+
+/**
+ * For when there's a new donation subscription.
+ *
+ * This will be fetched from a new donation, so we're hooking into the 'donation_new' dispatch.
+ */
+Data_Events::register_listener(
+	'newspack_data_event_dispatch_donation_new',
+	'donation_subscription_new',
+	function( $timestamp, $data ) {
+		if ( 'once' === $data['recurrence'] ) {
+			return;
+		}
+		return $data;
+	}
+);
+
+/**
+ * For when a WooCommerce Subscription is cancelled.
+ */
+Data_Events::register_listener(
+	'woocommerce_subscription_status_updated',
+	'donation_subscription_cancelled',
+	function( $subscription, $status_from, $status_to ) {
+		if ( 'cancelled' !== $status_to ) {
+			return;
+		}
+		return [
+			'subscription_id' => $subscription->get_id(),
+			'user_id'         => $subscription->get_customer_id(),
+			'email'           => $subscription->get_billing_email(),
+		];
+	}
+);
+
+/**
+ * For when a WooCommerce Subscription status changes.
+ */
+Data_Events::register_listener(
+	'woocommerce_subscription_status_updated',
+	'donation_subscription_changed',
+	function( $subscription, $status_from, $status_to ) {
+		return [
+			'subscription_id' => $subscription->get_id(),
+			'user_id'         => $subscription->get_customer_id(),
+			'email'           => $subscription->get_billing_email(),
+			'status_before'   => $status_from,
```

Comment:
Testing this step:

> 7. Now make a recurring donation (subscription) and confirm you see the same with: donation_subscription_new, donation_subscription_changed (switching from pending to active)

The `donation_subscription_changed` action is dispatched going from `active` to `pending` instead of `pending` to `active` as described above:

<img width="1769" alt="Screen Shot 2023-01-12 at 5 22 45 PM" src="https://user-images.githubusercontent.com/2230142/212208737-aae79240-2010-4faf-a526-9024a026dedd.png">

Again, not sure if it's a bug or not, but it does differ from the testing steps.

Reply:
It looks like the order of arguments here is off. According to [WC's docs](https://woocommerce.com/document/subscriptions/develop/action-reference/#section-2) for the `woocommerce_subscription_status_updated` action hook, the second argument is the "new" status (`status_to`) and the third is the old (`status_from`).

Reply:
Indeed! Fixed with 8671e7b6b5654da06a17bc4bb5f24c6bdf3ee0db

Reply:
Confirmed!